### PR TITLE
Change module syntax to ES6

### PR DESCRIPTION
**Pull request by dervan**
To allow easy interoperability with web module syntax was moved to ES6. Within that migration also `turtledove-js-api` package was extracted.

### CODE REVIEW

File: ad-network/ad-network.js [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
+import { extractContextSignals, getContextualAd } from './context-evaluation.js'
+import { ports, addresses } from '../config.js'
 
-const app = express()
-app.use(cors())
-app.use(bodyParser.json())
+const __dirname = path.resolve('./ad-network')
 
 /**
  * For given InterestGroup returns prepared InterestGroupAd, in the format described by TURTLEDOVE.
```
Comment (piwanczak) [inline ad-network/ad-network.js]: docstring probably needs an update to reflect change from a single object to an array of InteresGroupAd(s)

File: turtledove-server/content/static/js/render-ad.js [hunk 2]
```diff
@@ -34,11 +32,17 @@ function renderAd (ad) {
   about.style.width = '1em'
   about.style.height = '1em'
   about.style.textAlign = 'center'
-  const description = ad.type === 'contextual' ? '<p>This ad was chosen based on the context of the article you are currently in.</p>'
-    : `<p>This is an ${ad.type} ad. It was fetched for the group ${ad?.interestGroupSignals.name} from the site <a href="${ad?.interestGroupSignals?.owner}" target="_top">${ad?.interestGroupSignals?.owner}</a>.</p>`
+  let description
+  if (ad.type === 'contextual') {
+    description = '<p>This ad was chosen based on the context of the article you are currently in.</p>'
+  } else if (ad.type === 'none') {
```
Comment (piwanczak) [inline turtledove-server/content/static/js/render-ad.js]: How would you feel about setting up a default / error as a last else? 
if (ad.type==='contextua') { (...) 
} else if (ad.type === 'interest-group') { (...) 
} else { // implicit: none or other error
    description = '<p>An error during ad choosing occured.</p>'
}

File: turtledove-server/content/static/js/render-ad.js [hunk 3]
```diff
@@ -34,11 +32,17 @@ function renderAd (ad) {
...
+    description = '<p>An error during ad choosing occured.</p>'
+  } else {
+    description = `<p>This is an ${ad.type} ad. It was fetched for the group ${ad?.interestGroupSignals.name} from the site <a href="${ad?.interestGroupSignals?.owner}" target="_top">${ad?.interestGroupSignals?.owner}</a>.</p>`
+  }
   about.innerHTML = '<b>?</b><span class="tooltiptext">' +
     '<h4>Why I see this ad?</h4>' + description +
-    (ad.id != null ? `<b>Don't you like this ad?</b> Click <a href='/ad-remove?id=${ad.id}'>here</a> to delete.` : '') +
+    (ad.type === 'interest-group' ? `<b>Don't you like this ad?</b> Click <a href='/ad-remove?id=${ad.id}'>here</a> to delete.` : '') +
```
Comment (piwanczak) [inline turtledove-server/content/static/js/render-ad.js]: Isn't this 'if' redundant - assuming this path implicitly states that ad is indeed 'interest-group'?

**Review by piwanczak — COMMENTED** [review]
overall looking good, a few (minor) questions added

File: turtledove-server/content/static/js/render-ad.js (hunk 3 above)
Comment (dervan) [reply turtledove-server/content/static/js/render-ad.js]: Yes, now I can move it to the previous branch.

File: turtledove-server/content/static/js/render-ad.js (hunk 2 above)
Comment (dervan) [reply turtledove-server/content/static/js/render-ad.js]: I did a clearer switch there, it looks better.

File: ad-network/ad-network.js (hunk 1 above)
Comment (dervan) [reply ad-network/ad-network.js]: Done.

**Review by piwanczak — APPROVED** [review]

**Pull request merged**